photologue/forms.py

```diff
@@ -85,90 +55,12 @@ def clean(self):
     def save(self, request=None, zip_file=None):
         if not zip_file:
             zip_file = self.cleaned_data['zip_file']
-        zip = zipfile.ZipFile(zip_file)
-        count = 1
-        current_site = Site.objects.get(id=settings.SITE_ID)
-        if self.cleaned_data['gallery']:
-            logger.debug('Using pre-existing gallery.')
-            gallery = self.cleaned_data['gallery']
+        if USE_CELERY:
```

Comment:
I'm wondering if it might be more consistent to always save to the ZipUploadModel model instance - and to avoid the different behaviour between synchronous and asynchronous modes. It could also help to make the code shorter.